mllib/src/main/scala/org/apache/spark/ml/feature/StandardScaler.scala

```diff
@@ -189,7 +189,6 @@ object StandardScalerModel extends MLReadable[StandardScalerModel] {
         sqlContext.read.parquet(dataPath)
           .select("std", "mean", "withStd", "withMean")
           .head()
-      // This is very likely to change in the future because withStd and withMean should be params.
```

Comment:
If `withMean` and `withStd` are parameters, we should save them in `metadata/` but not both under `data/` and `medadata/`. Can we change the constructor of `ml.StandardScalerModel` to take only `std` and `mean` but construct `scaler` only inside `transform`? So `scaler` is no longer a member variable. We can fix performance issues in 1.7.